fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/retry/RetryPolicy.scala

```diff
@@ -0,0 +1,207 @@
+package izumi.functional.bio.retry
+
+import izumi.functional.bio.retry.RetryPolicy.{ControllerDecision, RetryFunction}
+import izumi.functional.bio.{F, Monad2}
+
+import java.time.ZonedDateTime
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.FiniteDuration
+
+abstract class RetryPolicy[F[+_, +_]: Monad2, -A, +B](val action: RetryFunction[F, A, B]) {
+  def &&[A1 <: A, B1](policy: RetryPolicy[F, A1, B1]): RetryPolicy[F, A1, (B, B1)] = {
+    def combined(left: RetryFunction[F, A, B], right: RetryFunction[F, A1, B1]): RetryFunction[F, A1, (B, B1)] =
+      (now: ZonedDateTime, in: A1) =>
+        F.map2(left(now, in), right(now, in)) {
+          case (ControllerDecision.Stop(l), ControllerDecision.Stop(r)) => ControllerDecision.Stop(l -> r)
+          case (ControllerDecision.Repeat(l, _, _), ControllerDecision.Stop(r)) => ControllerDecision.Stop(l -> r)
+          case (ControllerDecision.Stop(l), ControllerDecision.Repeat(r, _, _)) => ControllerDecision.Stop(l -> r)
+          case (ControllerDecision.Repeat(l, linterval, lnext), ControllerDecision.Repeat(r, rinterval, rnext)) =>
+            val combinedInterval = if (linterval.toInstant.compareTo(rinterval.toInstant) >= 0) linterval else rinterval
+            ControllerDecision.Repeat(l -> r, combinedInterval, combined(lnext, rnext))
+        }
+    RetryPolicy(combined(action, policy.action))
+  }
+
+  def ||[A1 <: A, B1](policy: RetryPolicy[F, A1, B1]): RetryPolicy[F, A1, (B, B1)] = {
+    def combined(left: RetryFunction[F, A, B], right: RetryFunction[F, A1, B1]): RetryFunction[F, A1, (B, B1)] = {
+      (now: ZonedDateTime, in: A1) =>
+        F.map2(left(now, in), right(now, in)) {
+          case (ControllerDecision.Repeat(l, linterval, lnext), ControllerDecision.Repeat(r, rinterval, rnext)) =>
+            val combinedInterval = if (linterval.toInstant.compareTo(rinterval.toInstant) < 0) linterval else rinterval
+            ControllerDecision.Repeat(l -> r, combinedInterval, combined(lnext, rnext))
+          case (ControllerDecision.Repeat(l, interval, next), ControllerDecision.Stop(r)) =>
+            ControllerDecision.Repeat(l -> r, interval, combined(next, RetryFunction.done(r)))
+          case (ControllerDecision.Stop(l), ControllerDecision.Repeat(r, interval, next)) =>
+            ControllerDecision.Repeat(l -> r, interval, combined(RetryFunction.done(l), next))
+          case (ControllerDecision.Stop(l), ControllerDecision.Stop(r)) => ControllerDecision.Stop(l -> r)
+        }
+    }
+    RetryPolicy(combined(action, policy.action))
+  }
+
+  def >>>[B1](that: RetryPolicy[F, B, B1]): RetryPolicy[F, A, B1] = {
```

Comment:
@VladPodilnyk Could you check if you could move `: Monad2`/`: Applicative2` constraints down to method level instead of the class header?

```suggestion
  def >>>[B1](that: RetryPolicy[F, B, B1])(implicit F: Monad2[F]): RetryPolicy[F, A, B1] = {
```

Reply:
Yup, it is possible. 
Done ✅